src/content/learning/using-durable-objects.md

```diff
@@ -0,0 +1,216 @@
+
+```js
+
+class DurableObjectExample {
+    constructor(state, environment){
+        this.state = state
+    }
+    
+    addEventListener("fetch", (event) => {
+        event.respondWith(handle(event.request));
+    });
+ 
+    async function handle(request) {
+        let ip = request.headers.get('CF-Connecting-IP');
+        let data = request.text();
+        let storagePromise = this.state.storage.set(ip, data);
+        await storagePromise;
+        return new Response(ip + ' stored ' + data);
+    }
+
+}
+
+```
+
+Single statement transactions are always transactional.  More complex use cases can wrap multiple storage statements in a transaction.  
+
+```js
+
+class DurableObjectExample {
+    constructor(state, environment){
+        this.state = state
+    }
+    
+    addEventListener("fetch", (event) => {
+        event.respondWith(handle(event.request));
+    });
+ 
+    async function handle(request) {
+        // TODO: txn example
+    }
+
+}
+
+```
+
+Transactions can fail if they conflict with a concurrent transaction from the same Durable Object.  Transactions are transparently and automatically retried once before returning a failure.
+
+
+## Binding the class to a Worker
+
+As with Workers KV, the class namespace must be created and then bound into a Worker before it can be used.
+
+First, upload the script that contains the class via the Workers Dashboard or via the command line.
+
+```sh
+curl -i -H "X-Auth-Email: ${EMAIL}" -H "X-Auth-Key: ${AUTH_KEY}" "https://api.cloudflare.com/client/v4/accounts/${ACCOUNT_TAG}/workers/scripts/${SCRIPT_NAME}" -X PUT -H "Content-Type: application/javascript" --data "`cat durable_object_example.js`"
+```
+
+To create the class, call the following endpoint.
+
+```sh
+curl -i -H "X-Auth-Email: ${EMAIL}" -H "X-Auth-Key: ${AUTH_KEY}" "https://api.cloudflare.com/client/v4/accounts/${ACCOUNT_TAG}/workers/durable_objects/namespaces" -X POST --data "{'name': 'example-class', 'script': ${SCRIPT_NAME}"
+```
+
+You can now create a binding for the Worker which uses this class.  TODO: Not sure if this is still the correct syntax, check.
+
+```js 
+// binding.json
+{
+  "body_part": "script",
+  "bindings": [
+    {
+      "type": "durable_object_namespace",
+      "name": "example-class",
+      "namespace_id": "<namespace id>"
+    }
+  ]
+}
+
+```
+
+
+## Instantiating a Durable Object
+
+Creating a Durable Object requires retrieving a Fetcher object from the class in your Workers Script.  Once you have the Fetcher object, you can use it to send [Requests](/runtime-apis/request) to the Durable Object instance.
+
+```js
+
+addEventListener('fetch', event => {
+  event.respondWith(handleRequest(event.request))
+})
+ 
+async function handleRequest(request) {
+  // Retrieve the Fetcher for the Durable Object with id 1
+  let durableObject = await DURABLE_OBJECT_CLASS.get("1")
```

Comment:
It'd be nice to make this binding name ("DURABLE_OBJECT_CLASS") match the "name" provided in the binding definition in the section above, since that's where it comes from.